Userland/DevTools/UserspaceEmulator/MallocTracer.h

```diff
@@ -39,6 +39,23 @@ namespace UserspaceEmulator {
 class Emulator;
 class SoftCPU;
 
+struct GraphNode {
+    Vector<FlatPtr> edges_from_node {};
+
+    bool is_reachable { false };
+};
+
+struct MemoryGraph {
+    HashMap<FlatPtr, GraphNode> map;
+
+    GraphNode get(FlatPtr address)
+    {
+        auto value_or_error = map.get(address);
+        VERIFY(value_or_error.has_value());
+        return value_or_error.value();
```

Comment:
Calling `Optional<T>::value()` does a VERIFY that there is a value, so I'm not sure that this helper gets you much at the moment. Seems you could replace this struct with a type alias until more graph-like operations on the Memory graph need specific member functions.

```
using MemoryGraph = HashMap<FlatPtr, GraphNode>:
```

Reply:
Ah, that makes a lot of sense!